src/BlockTypes/ProductGalleryLargeImage.php

```diff
@@ -0,0 +1,79 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+/**
+ * ProductGalleryLargeImage class.
+ */
+class ProductGalleryLargeImage extends AbstractBlock {
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name = 'product-gallery-large-image';
+
+	/**
+	 * It isn't necessary register block assets because it is a server side block.
+	 */
+	protected function register_block_type_assets() {
+		return null;
+	}
+
+	/**
+	 *  Register the context
+	 *
+	 * @return string[]
+	 */
+	protected function get_block_type_uses_context() {
+		return [ 'query', 'queryId', 'postId' ];
+	}
+
+	/**
+	 * Include and render the block.
+	 *
+	 * @param array    $attributes Block attributes. Default empty array.
+	 * @param string   $content    Block content. Default empty string.
+	 * @param WP_Block $block      Block instance.
+	 * @return string Rendered block type output.
+	 */
+	protected function render( $attributes, $content, $block ) {
+		$post_id = $block->context['postId'];
+
+		if ( ! isset( $post_id ) ) {
+			return '';
+		}
+
+		global $product;
+
+		$previous_product = $product;
+		$product          = wc_get_product( $post_id );
+		if ( ! $product instanceof \WC_Product ) {
+			$product = $previous_product;
+
+			return '';
+		}
+
+		if ( class_exists( 'WC_Frontend_Scripts' ) ) {
+			$frontend_scripts = new \WC_Frontend_Scripts();
+			$frontend_scripts::load_scripts();
+		}
+
+		ob_start();
+		remove_action( 'woocommerce_product_thumbnails', 'woocommerce_show_product_thumbnails', 20 );
```

Comment:
Considering that we are removing this action to prevent the thumbnails from appearing, do you think we should add it back after getting the gallery html (using the `add_action`)?

Reply:
Hmm, I'm not sure, perhaps this indeed would be the clean way to do it. That said, I can't really imagine a scenario in which someone would be using the blockified template and the Product Gallery block, and for some reason would like to utilize the `woocommerce_show_product_images()` again in a different context.

Reply:
I will look into this further :)

Reply:
I'm implementing the Thumbnails block and I'm using the following [code] (https://github.com/woocommerce/woocommerce-blocks/pull/10063/files#diff-865f98c619d1ca4f27d026d7c9bdd132a9d049158ff3b82701ed2e0426b42fc9R59) to get the thumbnails:

```php
  ob_start();
  do_action( 'woocommerce_product_thumbnails' );
  $product_gallery_thumbnails_html = ob_get_clean();
```

With that I can output the thumbnails, the only issue is that the styles are not being applied correctly (I'll investigate this further next week) but maybe we can use something like this instead of remove the action 🤔

Reply:
> I can't really imagine a scenario in which someone would be using the blockified template and the Product Gallery block, and for some reason would like to utilize the woocommerce_show_product_images() again in a different context.

I'm not sure if this is a possible scenario, but what if someone is using a plugin that uses this action that is being removed? Could it break the application?

Reply:
I am now re-adding the action.